utils/io/content/contentreader_test.go

```diff
@@ -30,71 +27,86 @@ type ArrayValue struct {
 	// Read the same file two times
 	for i := 0; i < 2; i++ {
 		var rSlice []inputRecord
-		var r inputRecord
-		var err error
-		for err = rr.NextRecord(&r); err == nil; err = rr.NextRecord(&r) {
-			rSlice = append(rSlice, r)
+		for item := new(inputRecord); reader.NextRecord(item) == nil; item = new(inputRecord) {
+			rSlice = append(rSlice, *item)
 		}
-		assert.Equal(t, err, io.EOF)
-		assert.NoError(t, rr.GetError())
+		assert.NoError(t, reader.GetError())
 		// First element
-		assert.Equal(t, rSlice[0].IntKey, 1)
-		assert.Equal(t, rSlice[0].StrKey, "A")
-		assert.Equal(t, rSlice[0].BoolKey, true)
+		assert.Equal(t, 1, rSlice[0].IntKey)
+		assert.Equal(t, "A", rSlice[0].StrKey)
+		assert.Equal(t, true, rSlice[0].BoolKey)
 		assert.ElementsMatch(t, rSlice[0].ArrayKey, []ArrayValue{{Key: "build.number", Value: "6"}})
 		// Second element
-		assert.Equal(t, rSlice[1].IntKey, 2)
-		assert.Equal(t, rSlice[1].StrKey, "B")
-		assert.Equal(t, rSlice[1].BoolKey, false)
+		assert.Equal(t, 2, rSlice[1].IntKey)
+		assert.Equal(t, "B", rSlice[1].StrKey)
+		assert.Equal(t, false, rSlice[1].BoolKey)
 		assert.Empty(t, rSlice[1].ArrayKey)
-		rr.Reset()
+		// Length validation
+		len, err := reader.Length()
+		assert.NoError(t, err)
+		assert.Equal(t, 2, len)
+		reader.Reset()
 	}
 }
 
 func TestContentReaderEmptyResult(t *testing.T) {
-	searchResultPath := filepath.Join(getTestDataPath(), "contentreaderwriter", emptySearchResult)
-	rr := NewContentReader(searchResultPath, arrayKey)
-	var r inputRecord
-	for e := rr.NextRecord(&r); e == nil; e = rr.NextRecord(&r) {
+	searchResultPath := filepath.Join(getTestDataPath(), emptySearchResult)
+	reader := NewContentReader(searchResultPath, DefaultKey)
+	for item := new(inputRecord); reader.NextRecord(item) == nil; item = new(inputRecord) {
 		t.Error("Can't loop over empty file")
 	}
-	assert.NoError(t, rr.GetError())
+	assert.NoError(t, reader.GetError())
 }
 
 func getTestDataPath() string {
 	dir, _ := os.Getwd()
-	return filepath.Join(dir, "..", "..", "..", "tests", "testsdata")
+	return filepath.Join(dir, "..", "..", "..", "tests", "testsdata", "contentreaderwriter")
 }
 
 func TestCloseReader(t *testing.T) {
 	// Create a file.
-	fd, err := ioutil.TempFile("", strconv.FormatInt(time.Now().Unix(), 10))
+	fd, err := fileutils.CreateTempFile()
 	assert.NoError(t, err)
 	fd.Close()
 	filePathToBeDeleted := fd.Name()
 
 	// Load file to reader
-	rr := NewContentReader(filePathToBeDeleted, arrayKey)
+	reader := NewContentReader(filePathToBeDeleted, DefaultKey)
 
 	// Check file exists
 	_, err = os.Stat(filePathToBeDeleted)
 	assert.NoError(t, err)
 
 	// Check if the file got deleted
-	err = rr.Close()
-	assert.NoError(t, err)
+	assert.NoError(t, reader.Close())
 	_, err = os.Stat(filePathToBeDeleted)
 	assert.True(t, os.IsNotExist(err))
 }
+
+func TestLengthCount(t *testing.T) {
+	searchResultPath := filepath.Join(getTestDataPath(), searchResult)
+	reader := NewContentReader(searchResultPath, DefaultKey)
```

Comment:
Looks like this reader is not closed.

Reply:
https://github.com/jfrog/jfrog-client-go/pull/166#discussion_r456895260